### PR TITLE
Fix/apcu memory with segments

### LINKED ISSUE #4: Wrong size for APCu
Hello Mittwald-Team,

I have configured shm_size to 128M, but your widget tolds me that I only have 32M configured.
You are reading seg_size which defaults to 32M on my OS that is correct, but you don't check for shm_segments which is configured to 4 on my system. 4 * 32M = 128M.

Stefan

### PR DESCRIPTION
**Pull request by dermodmaster**
Resolves #4

### CODE REVIEW

File: ext_emconf.php [hunk 1]
```diff
@@ -12,11 +12,7 @@
   'author_email' => 'support@mittwald.de',
   'author_company' => 'Mittwald CM Service GmbH',
   'state' => 'beta',
-<<<<<<< HEAD
-  'version' => '1.1.0',
-=======
-  'version' => '1.0.0',
->>>>>>> parent of 812eb1f... v1.10
+  'version' => '1.1.1',
```
Comment (martin-helmich) [inline ext_emconf.php:15]: Please raise the version separately (resolving the previous merge conflict 🤦 is fine, though)

File: Classes/Widgets/ApcuCacheWidget.php [hunk 2]
```diff
@@ -44,9 +44,10 @@ protected function loadData(): void
         {
             $apcuData = apcu_sma_info();
             $this->widgetEnabled = True;
-            $this->usedMemory = floatval(number_format(($apcuData["seg_size"] - $apcuData["avail_mem"])/1024/1024, self::decimals));
+            $total = $apcuData['num_seg']*$apcuData["seg_size"];
+            $this->usedMemory = floatval(number_format(($total - $apcuData["avail_mem"])/1024/1024, self::decimals));
             $this->freeMemory = floatval(number_format($apcuData["avail_mem"]/1024/1024, self::decimals));
-            $this->sumMemory = floatval(number_format($apcuData["seg_size"]/1024/1024, self::decimals));
+            $this->sumMemory = floatval(number_format($total/1024/1024, self::decimals));
```
Comment (martin-helmich) [inline Classes/Widgets/ApcuCacheWidget.php:50]: These are WAY to complicated in the first place. If all you want to do is round to two decimals, why not just use `round` with the `$precision` parameter?

    $this->usedMemory = round(($total - $apcuData["avail_mem"])/1024/1024, self::decimals);

File: Classes/Widgets/ApcuCacheWidget.php [hunk 3]
```diff
@@ -44,9 +44,10 @@ protected function loadData(): void
         {
             $apcuData = apcu_sma_info();
             $this->widgetEnabled = True;
-            $this->usedMemory = floatval(number_format(($apcuData["seg_size"] - $apcuData["avail_mem"])/1024/1024, self::decimals));
+            $total = $apcuData['num_seg']*$apcuData["seg_size"];
```
Comment (martin-helmich) [inline Classes/Widgets/ApcuCacheWidget.php:47]: Unrelated to this change, but nevertheless: `apcu_sma_info()` may return `false` on certain (not really documented) conditions. Please add a check for that condition.

Comment (martin-helmich) [inline Classes/Widgets/ApcuCacheWidget.php:47]: Also, formatting:

```suggestion
            $total = $apcuData['num_seg'] * $apcuData['seg_size'];
```

**Review by martin-helmich — CHANGES_REQUESTED** [review]

**Comment by derhansen** [comment]
Closes with 385ec1f and 65f4831

**Pull request closed without merging**